tests/utils.go

```diff
@@ -1567,3 +1567,38 @@ func GetNodeLibvirtCapabilities(nodeName string) string {
 
 	return output
 }
+
+// KubevirtFailHandler call ginkgo.Fail with printing the additional information
+func KubevirtFailHandler(message string, callerSkip ...int) {
+	virtClient, err := kubecli.GetKubevirtClient()
+	PanicOnError(err)
+
+	for _, ns := range []string{metav1.NamespaceSystem, NamespaceTestDefault} {
+		// Get KubeVirt specific pods information
+		pods, err := virtClient.CoreV1().Pods(ns).List(metav1.ListOptions{LabelSelector: "kubevirt.io"})
+		PanicOnError(err)
+
+		for _, pod := range pods.Items {
+			fmt.Printf("\nPod name: %s\t Pod phase: %s\n\n", pod.Name, pod.Status.Phase)
+			var tailLines int64 = 15
+			var containerName = ""
+			if strings.HasPrefix(pod.Name, "virt-launcher") {
+				containerName = "compute"
+			}
+			logsRaw, err := virtClient.CoreV1().Pods(metav1.NamespaceSystem).GetLogs(
+				pod.Name, &k8sv1.PodLogOptions{
+					TailLines: &tailLines,
+					Container: containerName,
+				},
+			).DoRaw()
+			if err == nil {
```

Comment:
If err != nil, that means the logs couldn't be fetched, right? Is that a normal situation? If not should we mention that we couldn't get them?

Reply:
@stu-gott I think `Fail` handler runs in separate go routine, so when you coming to fetch logs from `virt-launcher` pod, it possible that it already does not exists, but we just want to continue with the standard `Fail` handler flow so I do not fail on fetch logs problem.